### PR TITLE
PHPC-1017: Allow configure options to be specified with PECL install

### PR DESCRIPTION
**Pull request by derickr**
https://jira.mongodb.org/browse/PHPC-1017

Added the configurations to the ``package.xml.in`` file, and also align the options in ``./configure --help`` (it's been bugging me for a while).

### CODE REVIEW

File: bin/package.xml.in [hunk 1]
```diff
@@ -55,5 +55,11 @@ necessary to build a fully-functional MongoDB driver.
   </required>
  </dependencies>
  <providesextension>mongodb</providesextension>
- <extsrcrelease/>
+ <extsrcrelease>
+  <configureoption name="with-libbson" default="no" prompt="Use system libbson"/>
+  <configureoption name="with-libmongoc" default="no" prompt="Use system libmongoc"/>
+  <configureoption name="with-openssl-dir" default="yes" prompt="OpenSSL dir for MongoDB"/>
```
Comment (jmikola) [inline bin/package.xml.in]: Should we add a note that "yes" results in automatic detection?

File: bin/package.xml.in [hunk 2]
```diff
@@ -55,5 +55,11 @@ necessary to build a fully-functional MongoDB driver.
...
+  <configureoption name="with-system-ciphers" default="no" prompt="Use system default cipher list instead of hardcoded value"/>
```
Comment (jmikola) [inline bin/package.xml.in]: This may also warrant a note that it's for OpenSSL. Perhaps "OpenSSL: Use system default..."?

**Comment by alcaeus** [comment]
@jmikola so in the future, it's `yes | pecl install mongodb`? Good thing it's 24pullrequests time :)

File: bin/package.xml.in (hunk 1 above)
Comment (derickr) [reply bin/package.xml.in]: Using "yes" (the binary) would also mean: system libbson, system libmongoc, and system-chiphers. I don't think we should recommend that?

**Comment by jmikola** [comment]
> so in the future, it's `yes | pecl install mongodb`?

I think so. I recall having to do something similar when installing the legacy driver on Travis due to: https://github.com/mongodb/mongo-php-driver-legacy/blob/3321e9849a3d17503707ded728586ce34d69fb69/package.xml#L212-L214

**Comment by jmikola** [comment]
> Using "yes" (the binary) would also mean: system libbson, system libmongoc, and system-chiphers. I don't think we should recommend that?

@alcaeus: @derickr raised a good point above. I think we'll need to rework this a bit.

**Comment by derickr** [comment]
Won't Fix.

**Pull request closed without merging**